app/routes.py

```diff
@@ -1,2 +1,258 @@
-from flask import Blueprint
+from flask.wrappers import Response
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from flask import Blueprint, jsonify, make_response, request, abort
+from datetime import date
+import requests
+import os
+from dotenv import load_dotenv
 
+load_dotenv()
+
+# Creates Blueprints
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# Helper Functions
+
+def valid_int(number, parameter_type):
+    try:
+        int(number)
+    except:
+        abort(make_response({"error": f"{parameter_type} must be an int"}, 400))
+
+def get_task_from_id(task_id):
+    valid_int(task_id, "task_id")
+    return Task.query.get_or_404(task_id)
+
+def get_goal_from_id(goal_id):
+    valid_int(goal_id, "goal_id")
+    return Goal.query.get_or_404(goal_id)
+
+# Tasks Routes
+
+@tasks_bp.route("", methods=["GET"])
+def read_all_tasks():
+
+    sort_query = request.args.get("sort")
+
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title).all()
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all()
+    else:
+        tasks = Task.query.all()
+
+    response_body = [task.to_dict() for task in tasks]
+
+    return jsonify(response_body)
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        response_body = {
+            "details": "Invalid data"
+        }
+        return jsonify(response_body), 400
```

Comment:
You can directly return the response body:

```suggestion
        return jsonify({ "details": "Invalid data" }), 400
```